src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs

```diff
@@ -192,7 +192,17 @@ public async Task<IActionResult> AddConnection(ConnectionDto connection)
 
                 if (allChannelConnections.All(c => c.ClientId.ToString().Equals(clientId)))
                 {
-                    await _slackFacade.ArchiveChannel(connection.ChannelId.ToString());
+                    var channelsAll = await _slackFacade.GetConversations();
```

Comment:
Consider combining the lambdas and using a more expressive variable name, e.g. "unregisteredChannelHandles". Technically you could probably omit the ToList() call in the end and use variable.Count() in the if statement below. The performance hit for in-memory conversion is marginal, so its not a big deal. :)

Reply:
Check e9c5937 and 1094b00